tests/syscall/fs/host/host.c

```diff
@@ -3,59 +3,70 @@
 
 #if defined(_WIN32)
 #include <windows.h>
+#pragma comment(lib, "Shell32.lib")
+#else
+#include <dirent.h>
 #endif
 #include <openenclave/host.h>
 #include <openenclave/internal/syscall/host.h>
 #include <openenclave/internal/tests.h>
 #include <stdio.h>
+#include <string.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include "fs_u.h"
 
 #define SKIP_RETURN_CODE 2
 
-int rmdir(const char* path);
-
-int test_fs_posix(
-    const char* enclave_path,
-    const char* src_dir,
-    const char* tmp_dir)
+#if defined(_WIN32)
+int rmdir(wchar_t* path)
 {
-    const uint32_t flags = oe_get_create_flags();
-    if ((flags & OE_ENCLAVE_FLAG_SIMULATE))
+    int ret = -1;
+    wchar_t* doublenullpath = NULL;
+    int len = (int)wcslen(path);
+    SHFILEOPSTRUCTW opt;
+
+    doublenullpath = malloc((len + 2) * sizeof(wchar_t));
+    if (!doublenullpath)
     {
-        printf("=== Skipped unsupported test in simulation mode (sealKey)\n");
-        return SKIP_RETURN_CODE;
+        goto done;
     }
+    memcpy(doublenullpath, path, len * sizeof(wchar_t));
+    doublenullpath[len] = doublenullpath[len + 1] = L'\0';
```

Comment:
Can you explain why this double-null path is needed? It has not been my experience that any of the Win32 wchar APIs require this and the earlier use of `wcslen` already assumes that the input `path` is null-terminated.

Reply:
Yes double-null string is a requirement for this operation, someone explain here https://devblogs.microsoft.com/oldnewthing/20100218-00/?p=14893. But you are right wcslen already includes one null, I will check if one extra is good.

Reply:
Thanks for the pointer! https://docs.microsoft.com/en-us/windows/win32/api/shellapi/ns-shellapi-shfileopstructw does indeed call out the need for a double null-terminated string